### PR TITLE
Three.Legacy: Fire console warns only once.

### PR DESCRIPTION
**Pull request by pailhead**
When some method is deprecated, i tend to end up with a lot of warnings in the console. When it's something that fires a few times during init, i can live with it, when it's something in a loop i have to change it. 

This proposal aims to relieve some of that pressure to change from legacy right then and there when you see the warnings, since you most likely can't leave a warn printing in every frame. The user is notified of the deprecation, but the legacy method can proxy to the new method.

Something like this would also be easy to control with say `three.enableDeprecationWarnings =  true`.

### CODE REVIEW

**Comment by mrdoob** [comment]
> When some method is deprecated, i tend to end up with a lot of warnings in the console. When it's something that fires a few times during init, i can live with it, when it's something in a loop i have to change it.

Could you share specific cases you have had to deal with? We may just need to change a few instead of all.

**Comment by pailhead** [comment]
`addAttribute` 

Is something that i usually experience on some frames, when i load the app, The few logs that end up in the console i interpret as:
> I would like to kindly inform you of a deprecation.

`clearTarget`
Is something though that can happen every frame, the log in that case is a bit more harsh:
> You have to change this NOW!

:)

My suggestion though is to apply something like this in general, not just for a few cases. With the current HEAD i'm not sure which ones are the worst contenders, but moving from some RX to some RY can get nasty. Proxied like this, it's very easy to disable the warnings altogether.

**Comment by mrdoob** [comment]
I agree that `clearTarget` would benefit of this.

> My suggestion though is to apply something like this in general, not just for a few cases. 

I don't like methods that call methods that call methods "just in case", makes the code hard to follow.

> Proxied like this, it's very easy to disable the warnings altogether.

Ah... So this PR had a hidden purpose...

All that work put in the `Three.Legacy.js` file... 😥

You would rather hide the warnings than update your code. I think that is Babylon.js approach, maybe you're using the wrong library.

**Comment by pailhead** [comment]
I’m not sure I understood your comment nor the reference to Babylon.

**Comment by pailhead** [comment]
https://github.com/mrdoob/three.js/pull/16269#issuecomment-484416705

**Comment by pailhead** [comment]
Actually, i'd like to reopen this and hopefully get objective feedback if possible. 

I want to make an argument that `THREE.Legacy` stuff serves to sort of "cheat/hack" the semver system. 

Since three is being published on npm, it inevitably uses semver, but increments only the MINOR version, never the MAJOR.

Without the `THREE.Legacy` this would be invalid, as the interface would actually break. Hence, in order to have backwards compatibility, `Legacy` is crucial. 

However, i feel that `Legacy` comes in two flavors:
1. usable - you see a few warnings through out the life of the app
2. unusable - you get one if nor more warnings each frame you render

The usable, you can live with without updating your code. The unusuable may make development impossible, hence, it would be nice if these were fired only once.

**Comment by mrdoob** [comment]
Unrelated but related.

I've been considering  maybe not including `Legacy.js` in the main builds and create new builds with `Legacy.js`:

```
three.legacy.js
three.module.legacy.js
```

So these files will be only be used to help updating a project and once it's updated then the dev would use `three.js`, `three.min.js`, `three.module.js`, etc

That would also help with tree-shaking.

**Comment by pailhead** [comment]
There would be two different versions with each build? If you don't want to change your code, you use `.legacy`, vice versa, if you use non `.legacy` you have to update your code?

**Comment by mrdoob** [comment]
Hmm... if you don't want to update your code, why are you updating?

https://twitter.com/jonofyi/status/1262776252106244098

**Comment by pailhead** [comment]
Not sure if I understand. My point is that maybe you want to update to get a bug fix and that is the only expectation that you have. The scenario i'm thinking is, say 2 months into a project, and 2 weeks after implementing a feature, you run into a bug. 

Without `Legacy` you are compelled to change code (eg. `AxisHelper` -> `AxesHelper`). With `Legacy` you don't have to. 

My argument is that, with `Legacy` you are still compelled to change some, because of how often they are likely to fire.

**Comment by mrdoob** [comment]
I see...

Unfortunately that scenario doesn't sound compelling enough. I don't think not wanting to change `AxisHelper` to `AxesHelper` is a good enough reason to change the whole Three.Legacy file.

Also, we had something like this in 2014 already: https://github.com/mrdoob/three.js/pull/5791

**Comment by mrdoob** [comment]
Why don't you put this in your application code?

```js
( function () {

	var map = {};
	var warn = console.warn;

	console.warn = function ( message ) {

		if ( message in map ) return;

		warn( message );
		map[ message ] = true;

	};

} )();
```

**Comment by mrdoob** [comment]
I've [rted your tweet](https://twitter.com/pailhead011/status/1262902502380302337) about this issue but doesn't seem to be a problem that resonates with people.

I think I'll be better to add that code in your app instead of modifying the lib.

**Comment by pailhead** [comment]
This could work but I wouldn't want to modify native methods. There might be warnings unrelated to three which i would like not to include in this map.
